src/test/java/com/google/javascript/cl2dts/MultiFileTest.java

```diff
@@ -0,0 +1,30 @@
+package com.google.javascript.cl2dts;
+
+import static com.google.javascript.cl2dts.ProgramSubject.assertThatProgram;
+import static java.util.Collections.singletonList;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+
+public class MultiFileTest {
+
+  // Repro for https://github.com/angular/closure-to-dts/issues/101
+  @Test
+  public void testPruneSymbolsProvidedInOtherFile() throws Exception {
+    String expected = DeclarationGeneratorTests.getTestFileText(input("prune_other/require.d.ts"));
+    assertThatProgram(
+        singletonList(input("prune_other/require.js")),
+        singletonList(input("prune_other/provide.js")))
+        .generatesDeclarations(false, expected);
+  }
+
+  private File input(String relativePath) {
+    Path testDir = FileSystems.getDefault().getPath("src", "test", "java");
```

Comment:
On Fri, Sep 18, 2015 at 4:49 PM Martin Probst <notifications@github.com>
wrote:

> In src/test/java/com/google/javascript/cl2dts/MultiFileTest.java
> <https://github.com/angular/closure-to-dts/pull/102#discussion_r39911638>:
>
> > +import java.nio.file.Path;
> > +
> > +public class MultiFileTest {
> > +
> > +  // Repro for https://github.com/angular/closure-to-dts/issues/101
> > +  @Test
> > +  public void testPruneSymbolsProvidedInOtherFile() throws Exception {
> > +    String expected = DeclarationGeneratorTests.getTestFileText(input("prune_other/require.d.ts"));
> > +    assertThatProgram(
> > +        singletonList(input("prune_other/require.js")),
> > +        singletonList(input("prune_other/provide.js")))
> > +        .generatesDeclarations(false, expected);
> > +  }
> > +
> > +  private File input(String relativePath) {
> > +    Path testDir = FileSystems.getDefault().getPath("src", "test", "java");
>
> We could share the code to get the base directory, couldn't we?
>

Actually I think we might want to collapse the existing tests to share the
multi-file fixture, they just happen to have a single .js file.
Maybe in a followup change, this is getting big


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/closure-to-dts/pull/102/files#r39911638>.
>